Swap internal RPC server for API server in the helm chart

ashb opened the pull request:
Previously the PRC server was possible to disable/enable, where as now the new
api-server is a required component in Airflow 3.0+

This also does a little bit of drive-by refactoring of the helper templates
for service account name generation

I'm not 100% sure what "default" version is the tests, so maybe me adding a parameterization for "3.0.0" is not right. I also don't know what version we set it the kube/helm tests we run against main etc. 

Part of https://github.com/apache/airflow/issues/44436 (and needed to fix main after #44427 was merged)

<!--
 Licensed to the Apache Software Foundation (ASF) under one
 or more contributor license agreements.  See the NOTICE file
 distributed with this work for additional information
 regarding copyright ownership.  The ASF licenses this file
 to you under the Apache License, Version 2.0 (the
 "License"); you may not use this file except in compliance
 with the License.  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing,
 software distributed under the License is distributed on an
 "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 KIND, either express or implied.  See the License for the
 specific language governing permissions and limitations
 under the License.
 -->

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->



<!-- Please keep an empty line above the dashes. -->
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

ashb committed:
Swap internal RPC server for API server in the helm chart

Previously the PRC server was possible to disable/enable, where as now the new
api-server is a required component in Airflow 3.0+

This also does a little bit of drive-by refactoring of the helper templates
for service account name generation

kaxil reviewed (commented):
I think the failures are just in `helm_tests/airflow_core/test_rpc_server.py`

amoghrajesh commented on chart/templates/api-server/api-server-deployment.yaml:
```diff
@@ -18,44 +18,43 @@
-{{- $containerSecurityContext := include "containerSecurityContext" (list . .Values._rpcServer) }}
-{{- $containerSecurityContextWaitForMigrations := include "containerSecurityContext" (list . .Values._rpcServer.waitForMigrations) }}
-{{- $containerLifecycleHooks := or .Values._rpcServer.containerLifecycleHooks .Values.containerLifecycleHooks }}
+{{- if semverCompare ">=3.0.0" .Values.airflowVersion }}
```
Should be OK

ashb committed:
Mark the LocalExecutor test that uses example_xcom dag as xfail for now

We merged a change to swap the LocalExecutor over to the new AIP-72 TaskSDK
code, and it wasn't complete enough to work with xcom. To unblock main we are
temporarirly marking that test as xfail, with a link to the issue to fix it.

ashb commented on kubernetes_tests/test_other_executors.py:
```diff
@@ -54,6 +54,7 @@ def test_integration_run_dag(self):
             timeout=300,
         )
 
+    @pytest.mark.xfail(reason="https://github.com/apache/airflow/issues/44481 needs to be implemented")
```
Please note, I've marked one test as XFail for now.
